### ISSUE TITLE
Why we not use the full image url as cache key?

### ISSUE THREAD

**Issue by chriszyyy** [bubkoo/html-to-image#206]
The logic of caching image url is not working as expected
See here the source code of the issue https://github.com/bubkoo/html-to-image/blob/master/src/getBlobFromURL.ts#:~:text=let%20key%20%3D%20url.replace(/%5C%3F.*/%2C%20%27%27)

```js
function getCacheKey(url: string) {
  let key = url.replace(/\?.*/, '')

  // font resourse
  if (/ttf|otf|eot|woff2?/i.test(key)) {
    key = key.replace(/.*\//, '')
  }

  return key
}
```

If the image use query string to distinguish which image to get, that image all same format image with used the wrong cached one.

For example:
1. https://www.abc.com/images?id=123
2. https://www.abc.com/images?id=124

The second one will render with the first image because of the cache logic

### Expected Behavior

We should use the full URL as key to cache.

### Current Behavior

It only use the URL with domain and path, not query string.

### Possible Solution

Remove `let key = url.replace(/\?.*/, '')`

### Your Environment

- html-to-image: 1.9.0
- OS: Win10
- Browser: chrome

**Comment by al1khavari** [comment]
i have the same issue with this condition

**Comment by hallowatcher** [comment]
This is so underrated. I had to build my own version of this package in order to comment out the caching section.

I would've proposed a merge request but the build is failing for me locally, and the tests are failing. ESlint is also complaining about the hasOwnProperty calls and the length attribute for booleans

**Comment by al1khavari** [comment]
> This is so underrated. I had to build my own version of this package in order to comment out the caching section.
> 
> I would've proposed a merge request but the build is failing for me locally, and the tests are failing. ESlint is also complaining about the hasOwnProperty calls and the length attribute for booleans

I had this issues too , but i deciced to fork this package and build my own version also i fixed the eslint errors to build the package without any test be failed

**Comment by sweepline** [comment]
I also have this issue.
Can we do something like this in `getBlobFromURL` @bubkoo ?
```js
export function getBlobFromURL(
  url: string,
  options: Options,
): Promise<Metadata> {
  const cacheKey = options.cacheQueryString ? url : getCacheKey(url)
```

I currently have to run a patched version to use this as I'm using this with openlayers.

**Comment by sweepline** [comment]
its still an issue.

**Comment by logankuzyk** [comment]
This has been resolved in #260, set the `includeQueryParams` option.
